### PR TITLE
ASA-6609 & ASA-6586 (#88)

### PR DESCRIPTION
**Pull request by vishalhcl-5960**
* ASA-6609

* ASA-6609

* Update SAClient.java

* Update ServiceUtil.java

### CODE REVIEW

File: src/main/java/com/hcl/appscan/sdk/scanners/sast/SAClient.java [hunk 1]
```diff
@@ -112,8 +112,8 @@ private int runClient(String workingDir, List<String> args, String irGenClient,
                     String server = "-DBLUEMIX_SERVER="+serverURL;
                     if(acceptInvalidCerts != null && acceptInvalidCerts.equals("true")){
                         server = server+" -Dacceptssl";
-                        m_builder.environment().put("APPSCAN_OPTS",server);
                     }
+                    m_builder.environment().put("APPSCAN_OPTS",server);
```
Comment (mattmurp) [inline src/main/java/com/hcl/appscan/sdk/scanners/sast/SAClient.java:116]: I believe setting APPSCAN_OPTS here will override any settings that are in the environment variable. Is this necessary? If so, we may have to check for the existence of the variable and append the additional value if it exists.

File: src/main/java/com/hcl/appscan/sdk/utils/ServiceUtil.java [hunk 2]
```diff
@@ -87,8 +91,13 @@ public static String getSAClientVersion() throws IOException {
 	 * @return The current version of the package.
 	 * @throws IOException If an error occurs.
 	 */
-	public static String getSAClientVersion(Proxy proxy) throws IOException {
-		String request_url = SystemUtil.getDefaultServer() + String.format(API_SACLIENT_VERSION, API_SCX, SystemUtil.getOS(), "true"); //$NON-NLS-1$
+	public static String getSAClientVersion(Proxy proxy, String serverURL) throws IOException {
+        String request_url;
+        if(serverURL != null && !serverURL.isEmpty() && !serverURL.contains("appscan.com")){
```
[the remark this answers is not in the document]
Comment (vishalhcl-5960) [reply src/main/java/com/hcl/appscan/sdk/utils/ServiceUtil.java]: Made the required changes...doesn't using the "appscan.com" check.

File: src/main/java/com/hcl/appscan/sdk/utils/ServiceUtil.java [hunk 3]
```diff
@@ -87,8 +91,13 @@ public static String getSAClientVersion() throws IOException {
...
+            request_url = serverURL + String.format(API_SACLIENT_VERSION, API_SCX, SystemUtil.getOS(), "true"); //$NON-NLS-1$
```
[the remark this answers is not in the document]
Comment (vishalhcl-5960) [reply src/main/java/com/hcl/appscan/sdk/utils/ServiceUtil.java]: Made the changes.

File: src/main/java/com/hcl/appscan/sdk/scanners/sast/SAClient.java (hunk 1 above)
Comment (vishalhcl-5960) [reply src/main/java/com/hcl/appscan/sdk/scanners/sast/SAClient.java]: Yes Matt, It is Necessary. I made the required changes according to your comments.

**Review by mattmurp — APPROVED** [review]

**Pull request merged**